tests/experimental/datasources/test_spark_datasource.py

```diff
@@ -0,0 +1,275 @@
+import logging
+import pathlib
+from dataclasses import dataclass
+from typing import List
+
+import pytest
+
+import great_expectations.exceptions as ge_exceptions
+from great_expectations.alias_types import PathStr
+from great_expectations.data_context.util import file_relative_path
+from great_expectations.experimental.datasources.interfaces import (
+    BatchSortersDefinition,
+)
+from great_expectations.experimental.datasources.spark_datasource import (
+    CSVSparkAsset,
+    SparkDatasource,
+)
+
+LOGGER = logging.getLogger(__name__)
+
+
+@pytest.fixture
+def spark_datasource() -> SparkDatasource:
+    return SparkDatasource(name="spark_datasource")
+
+
+@pytest.fixture
+def csv_path() -> pathlib.Path:
+    return pathlib.Path(
+        file_relative_path(
+            __file__,
+            pathlib.Path("..", "..", "test_sets", "taxi_yellow_tripdata_samples"),
+        )
+    )
+
+
+@pytest.mark.unit
```

Comment:
If using a true Spark connection, I don't believe these are unit tests?

Reply:
@cdkini I am not sure whether or not using the `pyspark` module in our local (non-distributed) tests (even in Azure) qualifies as a "true" connection.  So I am inclined to keep as is.  Thanks.

Reply:
I don't know the intricacies of this new datasource (or ZEP in general) so I'm going to defer to Gabe and Bill - I want to say these should be integration though

Reply:
The very requirement on connecting to Spark (even if that connection is not used) makes me think that unit is not applicable. Since we're taking a lot of care in crafting our unit tests, let's be confident in our approach here and be consistent

Reply:
@cdkini @Kilo59 @billdirks -- Fine with me either way; I would just want to appeal for consistency (if makes sense).  If this test is integration, then so should be the same tests for the other backends.  Thanks.

Reply:
I'm fine with changing it.
This is just an old `pylint` habit.
https://pylint.pycqa.org/en/latest/user_guide/messages/convention/invalid-name.html#invalid-name-c0103

Reply:
I don't believe any of these tests actually need the spark backend. We might instantiate one when the datasource comes up but the tests check the `Batch` (a dataclass of data describing the data in the batch) which might involve listing the csv files in a directory. We could mock the spark execution engine to more strongly separate this from using actual spark. For SQL tests, we do mock out the sql execution engine. For pandas everything is in memory and there is no external service to mock out. I believe you can use `execution_engine_override` as a hook to use a fake execution engine.